src/components/RequestError.js

```diff
@@ -48,6 +56,9 @@ class RequestErrorModal extends React.Component {
    */
   handleChangeServer = () => {
     $('#requestErrorModal').modal('hide');
+    // When user decides to change the server when an error happened, we must reset the loading information
+    this.props.dataLoaded({addressesFound: 0, transactionsFound: 0});
```

Comment:
Shouldn't it be done by the loading screen itself? Maybe in the error callback of a promise?

Reply:
Hummm, I think you are right. I should do in the reject of the promise that requests the addressesHistory, what do you think? 

Maybe in this line: https://github.com/HathorNetwork/hathor-wallet/blob/master/src/utils/wallet.js#L164

What do you think?

Reply:
I think the best place would be where the transaction is generated. I think that the error screen must show up, and the download of the address history will be restarted.

Maybe in this line: https://github.com/HathorNetwork/hathor-wallet/blob/master/src/utils/wallet.js#L251

What do you think?

Reply:
You are 100% correct! I was thinking wrongly about the promise chain. Will do this